tcconfig/shaper/htb.py

```diff
@@ -112,10 +133,26 @@ def _add_rate(self):
         ]
 
         if bandwidth != upper_limit_rate:
+            # This sets the burst/cburst values to the default tc tries to set,
+            # but works around a bug in tc where values are rounded incorrectly.
+
+            # Note that "tc class show" will still show burst that are too small,
+            # due to the inverse bug, but if you use "tc -r class show" you will see
+            # the actual size of the cbuffer in bytes.
+            mtu = 1600
+            rate = bandwidth.byte_per_sec
+            desired_burst = int(rate / get_hz() + mtu)
+
+            burst = bisect_left(
+                range(1 << 32),
+                True,
+                key=lambda b: calc_xmitsize_true(rate, calc_xmittime_bug(rate, b)) >= desired_burst,
```

Comment:
According to the official documentation, `key` was added to `bisect_left` in Python 3.10 and later.
https://docs.python.org/3.13/library/bisect.html#bisect.bisect_left
Is there a backport package or something?

If there is no other solution, dropping Python 3.9 support may be an option.
The EOL of Python 3.9 is just a few months away.

Reply:
I don't know of any backport.  I could fall back to my initial implementation's linear search algorithm for Python 3.9 (I should be able to do some math to prove that the loop will always halt within not too many steps).

Reply:
I've added code to fall back to linear search for Python < 3.10, and bounded the search at 2^32 even though it'll definitely cap out at much less than that for all but ridiculously huge bandwidths.